dvc/data/stage.py

```diff
@@ -336,6 +367,9 @@ def stage(
         logger.debug("staged tree '%s'", obj)
         if name != "md5":
             obj = _stage_external_tree_info(odb, obj, name)
+
+        untyped = HashInfo(obj.name, obj.hash_info.value.replace(".dir", ""))
+        odb.add(obj.fs_path, obj.fs, untyped)
```

Comment:
```suggestion
        odb.add(obj.fs_path, obj.fs, obh.hash_info.as_raw())
```

Reply:
Side note: this `as_raw` is obviously not really about HashInfo, but more about ObjectID(Oid) that should be based on HashInfo. `HashInfo.isdir` is also clearly an oid thing 🙂 We'll introduce oids properly a bit later, clearly out of scope for this PR.